spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -131,12 +133,12 @@ public SparkTable(Table icebergTable, boolean refreshEagerly) {
   public SparkTable(Table icebergTable, String branch, boolean refreshEagerly) {
     this(icebergTable, refreshEagerly);
     this.branch = branch;
-    ValidationException.check(
-        branch == null
-            || SnapshotRef.MAIN_BRANCH.equals(branch)
-            || icebergTable.snapshot(branch) != null,
-        "Cannot use branch (does not exist): %s",
-        branch);
+    if (SnapshotRef.MAIN_BRANCH.equals(branch)) {
```

Comment:
If `branch` is main, I think we don't need to set `this.branch` as it's logically same as invoking `SparkTable(Table, boolean)`. However, some WAP tests require this.branch to be set for main.